gunicorn/arbiter.py

```diff
@@ -473,6 +473,8 @@ def manage_workers(self):
         Maintain the number of workers by spawning or killing
         as required.
         """
+        orig_num_workers = self.num_workers
+
         if len(self.WORKERS.keys()) < self.num_workers:
```

Comment:
What if `kill_worker` moved the worker from `self.WORKERS` to a dead worker list? I don't think `ESRCH` block is necessary because `reap_workers` should still get the `SIGCHLD`.

Then the `manage_workers` becomes three steps: